pkg/apis/tensorflow/validation/validation.go

```diff
@@ -19,10 +19,21 @@ import (
 	"fmt"
 
 	tfv1 "github.com/kubeflow/tf-operator/pkg/apis/tensorflow/v1alpha1"
+	tfv2 "github.com/kubeflow/tf-operator/pkg/apis/tensorflow/v1alpha2"
 	"github.com/kubeflow/tf-operator/pkg/util"
 )
 
-// ValidateTFJobSpec checks that the TFJobSpec is valid.
+// ValidateTFJobSpec checks that the v1alpha2.TFJobSpec is valid.
+func ValidateAlpha2TFJobSpec(c *tfv2.TFJobSpec) error {
+	for _, value := range c.TFReplicaSpecs {
+		if &value.Template == nil || len(value.Template.Spec.Containers) == 0 {
```

Comment:
No need to add & here.

Reply:
remove & will generate an error “Cannot contert nil to type v1.PodTemplateSpec”